src/_guides/language/concurrency/index.md

```diff
@@ -312,107 +311,144 @@ to implement isolates.
 
-[`first`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/Stream/first.html
+`_readAndParseJson()` is an existing,
+asynchronous function that could just as easily
+run directly in the main isolate.
+But using `Isolate.run()` enables concurrency,
+so its computations are completely abstracted by the worker isolate
+and can complete without blocking the main isolate.
 
-The spawned isolate executes the following code:
+Because `Isolate.run()` allows code in the main isolate to continue running
+while its computations execute in a new one,
+its result is always returned as a `Future`.
+The original isolate doesn't care whether the computation
+running in the other isolate is synchronous or asynchronous,
+because it's being run concurrently either way.
 
-<?code-excerpt "lib/simple_worker_isolate.dart (spawned)"?>
+{% comment %}
+TODO:
+Should create a diagram for the current example.
+Previous example's diagram and text for reference:
+
+  The following figure illustrates the communication between
+  the main isolate and the worker isolate:
+  
+  ![A figure showing the previous snippets of code running in the main isolate and in the worker isolate](/guides/language/concurrency/images/isolate-api.png)
+{% endcomment %}
+
+#### Sending closures with isolates
+
+This section shows another simple worker isolate created with `run()`,
+this time written as a function literal, or closure,
+directly in the main isolate.
+
+<?code-excerpt "lib/simple_isolate_closure.dart (main)"?>
 ```dart
-Future<void> _readAndParseJson(SendPort p) async {
-  final fileData = await File(filename).readAsString();
-  final jsonData = jsonDecode(fileData);
-  Isolate.exit(p, jsonData);
+void main() async {
+  // Read some data.
+  final jsonData = await Isolate.run(() async {
+    final fileData = await File(filename).readAsString();
+    final jsonData = jsonDecode(fileData) as Map<String, dynamic>;
+    return jsonData;
+  });
+
+  // Use that data.
+  print('Number of JSON keys: ${jsonData.length}');
 }
 ```
 
-The relevant statement is the last one, which exits the isolate,
-sending `jsonData` to the passed-in `SendPort`.
-Message passing using `SendPort.send` normally involves data copying,
-and thus can be slow.
-However, when you send the data using `Isolate.exit()`,
-then the memory that holds the message in the exiting isolate isn’t copied,
-but instead is transferred to the receiving isolate.
-The sender will nonetheless perform a verification pass to ensure
-the objects are allowed to be transferred.
-
-
-{{site.alert.version-note}}
-  `Isolate.exit()` was added in 2.15.
-  Previous releases support only explicit message passing,
-  using `Isolate.send()` as shown in the next section's example.
-{{site.alert.end}}
-
-The following figure illustrates the communication between
-the main isolate and the worker isolate:
+This example accomplishes the same as the previous:
+a new isolate spawns and executes a computation, sending back the result.
```

Comment:
**suggestion:** Avoid a compound sentence with a colon. If you can use a colon, you can split the sentence.
```suggestion
This example accomplishes the same as the previous.
A new isolate spawns, executes code, and returns the result.
```